test/fixtures/permission-validation/setup.js

```diff
@@ -0,0 +1,482 @@
+  const roles = {};
+  const createSubsystemRoles = async subsystem => {
+    const subsystemRoles = {};
+
+    await Promise.all(
+      rawRoles.map(async role => {
+        const { Name: name } = role;
+        role = {
+          ...role,
+          Name: name + subsystem,
+          Subsystem: subsystems[subsystem],
+        };
+        subsystemRoles[name] = await pgProvider.create('Role', role);
+      })
+    );
+
+    roles[subsystem] = subsystemRoles;
+  };
+
+  await Promise.all(subsystemNames.map(createSubsystemRoles));
+  return roles;
+};
+
+const prepareUsers = async subsystems => {
+  const users = {};
+  const createSubsystemUsers = async subsystem => {
+    const subsystemUsers = {};
+
+    await Promise.all(
+      rawUsers.map(async user => {
+        const { Login: login } = user;
+        user = {
+          ...user,
+          Login: login + subsystem,
+          Subsystem: subsystems[subsystem],
+        };
+        subsystemUsers[login] = await pgProvider.create('SystemUser', user);
+      })
+    );
+
+    users[subsystem] = subsystemUsers;
+  };
+
+  await Promise.all(subsystemNames.map(createSubsystemUsers));
+  return users;
+};
+
+const getEntitiesIds = async () => {
+  const entities = ['Action', 'Category', 'Application'];
+  return Promise.all(
+    entities.map(async category => {
+      const res = await pgProvider.select(category, {}).fetch();
+      return res.reduce((ids, row) => {
+        ids[row.Name] = row.Id;
+        return ids;
+      }, {});
+    })
+  );
+};
+
+const prepareCatalogs = async categories => {
+  const catalogs = {};
+
+  await Promise.all(
+    rawCatalogs.map(async catalog => {
+      catalog.Category = categories[catalog.Category];
+      catalogs[catalog.Caption] = await pgProvider.create('Catalog', catalog);
+    })
+  );
+
+  return catalogs;
+};
+
+const prepareTestData = async (categories, catalogs, subsystems) => {
+  const data = {};
+  const createGSObjects = async ([category, rawObjects]) => {
+    const objectIds = {};
+
+    await Promise.all(
+      rawObjects.map(async obj => {
+        const name = obj.Name || obj.Caption;
+        if (obj.Catalog) obj.Catalog = catalogs[obj.Catalog];
+        if (obj.Category) obj.Category = categories[obj.Category];
+        if (obj.Subsystem) obj.Subsystem = subsystems[obj.Subsystem];
+        objectIds[name] = await pgProvider.create(category, obj);
+      })
+    );
+
+    data[category.toLowerCase()] = objectIds;
+  };
+
+  await Promise.all(Object.entries(rawData).map(createGSObjects));
+  return data;
+};
+
+const preparePermissions = async (subsystems, categories, catalogs, roles) => {
+  const permissions = {};
+  permissions.S1 = {};
+  permissions.S2 = {};
+
+  const createRolePermissions = async ([role, rolePermissions]) => {
+    const preparedPermissions = [];
+
+    for (const permission of rolePermissions) {
+      for (const subsystem of subsystemNames) {
+        const preparedPermission = {};
```

Comment:
You can move the variable declarations after the if clause with the `continue` statement below, to improve readability and avoid unused variable creation.